doc/en/fixture.rst

```diff
@@ -804,48 +804,61 @@ without declaring a function argument explicitly or a `usefixtures`_ decorator.
 As a practical example, suppose we have a database fixture which has a
 begin/rollback/commit architecture and we want to automatically surround
 each test method by a transaction and a rollback.  Here is a dummy
-self-contained implementation of this idea::
+self-contained implementation of this idea in a style resembling xUnit setup
+and tearDown methods::
 
-    # content of test_db_transact.py
+    # content of test_foo.py
 
     import pytest
+    import MyFoo
 
-    class DB(object):
-        def __init__(self):
-            self.intransaction = []
-        def begin(self, name):
-            self.intransaction.append(name)
-        def rollback(self):
-            self.intransaction.pop()
+    foo = MyFoo()
 
-    @pytest.fixture(scope="module")
-    def db():
-        return DB()
+    class TestFoo:
+        @classmethod
+        @pytest.fixture(scope='class', autouse=True)
+        def setup(cls, request):
+            cls.session = foo.session()
+            def tearDown():
+                cls.session.logout()
+            request.addfinalizer(tearDown)
 
-    class TestClass(object):
-        @pytest.fixture(autouse=True)
-        def transact(self, request, db):
-            db.begin(request.function.__name__)
-            yield
-            db.rollback()
+        def test_foo(self):
+            session = self.session
+            # test code can now use session
+
+However, it is preferred to treat the test class purely as a container when
+using fixtures, so it makes more sense to stay away from storing state on it.
+Another way of realizing this, also using fixture names which do not conflict
+with xUnit or nose setup/tearDown compatibility names to avoid confusion::
+
+    # content of test_bar.py
+
+    import pytest
+    import MyFoo
+
+    foo = MyFoo()
 
-        def test_method1(self, db):
-            assert db.intransaction == ["test_method1"]
+    class TestFoo:
+        @classmethod
+        @pytest.fixture(scope='class', autouse=True)
+        def session(self, request):
+            session = foo.session()
+            request.addfinalizer(session.logout)
+            return session
 
-        def test_method2(self, db):
-            assert db.intransaction == ["test_method2"]
+        def test_foo(self):
+            # test code not using the session explicitly,
+            # will still be created and logged out however.
 
-The class-level ``transact`` fixture is marked with *autouse=true*
+        def test_bar(self, session):
+            # test code using session explicitly.
+
+The class-level ``session`` fixture is marked with *autouse=true*
 which implies that all test methods in the class will use this fixture
 without a need to state it in the test function signature or with a
 class-level ``usefixtures`` decorator.
 
-If we run it, we get two passing tests::
```

Comment:
Hmm I see you removed the part about running the test itself, so you may ignore my comment about `tox -e regen`. 👍